internals/overlord/planstate/manager.go

```diff
@@ -253,6 +256,10 @@ func (m *PlanManager) updatePlanLayers(layers []*plan.Layer) (*plan.Plan, error)
 	if err != nil {
 		return nil, err
 	}
+	newWorkloads, ok := p.Sections[workloads.WorkloadsField].(*workloads.Workloads)
+	if ok && !reflect.DeepEqual(newWorkloads, m.plan.Sections[workloads.WorkloadsField]) {
+		return nil, errors.New("plan workloads cannot be changed")
+	}
```

Comment:
This can work - just wondering if there is not a way to hide this completely in the workloads extension system. I have an idea ...